_posts/2017-12-11-A-note-on-fees.md

```diff
@@ -0,0 +1,118 @@
+---
+layout: post
+title: A note on fees
+tags: [crypto, research]
+author: dEBRUYNE / ArticMine
+---
+
+Lately, a common reoccurring complaint has been that fees are too expensive. Whilst we don't disagree with that statement, we have to thoroughly analyze the situation first. Furthermore, the notion of devs having to release new binaries with lower fees is myopic, because (i) it'd merely kick the can down the road, (ii) changing the constants or formulas requires a hard fork, i.e., they are enforced on a consensus level, and (iii) constantly intervening would be contradictory to our grass-roots, decentralized nature.  
+
+Let's start by comparing Monero's per kB fees to the per kB fees of other (hybrid) proof-of-work coins. Fees per kB for a typical transaction (2 inputs + 2 outputs):  
+
+- **Bitcoin:** ~$26.90  
+- **Ethereum:** ~$2.91  
+- **Litecoin:** ~$0.10  
+- **Dash:** ~$0.07  
+- **Monero:** ~$0.24  
+
+As you can see, the per kB fee of Monero is fairly low. However, due to the high transaction size, the absolute default fee (in $ terms) is quite high. Note that the transaction size is this big due to Monero's inherent default privacy, i.e., the range proofs, which mask the amount values, make up ~12 kB of a single transaction. RingCT, however, was absolutely necessary to strengthen the privacy of the network. More specifically, there were a lot of privacy "leaks" when Monero didn't mask amounts yet. Fortunately, [Bulletproofs](https://getmonero.org/2017/12/07/Monero-Compatible-Bulletproofs.html) will reduce transaction sizes by at least 80%.  
+
+-----------------
+
+To thoroughly analyze the situation, let's continue with examining the constants. We start with examining the penalty function and the dynamic block size algorithm. The formula is as follows:  
+
+**Penalty = BaseReward * ((BlockSize / M<sub>N</sub>) - 1)²**  
+
+The new reward is:  
+
+**NewReward = BaseReward - Penalty**  
+
+Where:  
+
+ -  M<sub>N</sub> is the median of the block size over the last N blocks, with N being [100](https://github.com/monero-project/bitmonero/blob/master/src/cryptonote_config.h#L57) in Monero  
+ -  BlockSize is the size of the current block  
+ -  BaseReward is the reward as per the emission curve or where applicable the tail emission  
+ -  NewReward is the actual reward paid to the miner  
+ -  The maximum allowed block size is 2M<sub>N</sub>  
+
+Note that the formula of the BaseReward is defined [as follows](https://bitcointalk.org/index.php?topic=583449.0):  
+
+**BaseReward = 2 * ((S - A) * 2<sup>-20</sup> * 10<sup>-12</sup>)**  
+
+Where:  
+
+ -  2 is the adjustment factor for the switch to two minute blocks  
+ -  S is the initial number of atomic units is = 2<sup>64</sup> - 1  
+ -  A is the current circulation, which can be found [here](https://moneroblocks.info/)  
```

Comment:
You could add here that the number of the current circulation (emission) on moneroblocks has to be multiplied with 10^12 in order to comply with S being in atomic units, or briefly explain atomic units in the definition of S. It took me (as a beginner) some time to figure out why my calculated BaseReward (with A = 15449232) was ~35 and not ~5.7XMR.

Reply:
Thanks for the suggestion. Added another line and squashed.